packages/test-sync/src/androidTest/kotlin/io/realm/kotlin/test/mongodb/shared/ApiKeyAuthTests.kt

```diff
@@ -0,0 +1,160 @@
+        val key1 = provider.create("foo")
+        val key2 = provider.fetch(key1.id)
+        assertEquals(key1.id, key2.id)
+        assertEquals(key1.name, key2.name)
+        assertNull(key2.value)
+        assertEquals(key1.enabled, key2.enabled)
+    }
+
+    @Test
+    fun fetch_nonExistingKeyThrows(): Unit = runBlocking {
+        // worth creating a more specific exception?
+        assertFailsWith<ServiceException> {
+            provider.fetch(ObjectId.create())
+        }
+    }
+
+    @Test
+    fun fetchAll() = runBlocking {
+        val key1 = provider.create("foo")
+        val key2 = provider.create("bar")
+        val keys = provider.fetchAll()
+        assertEquals(2, keys.size)
+        assertTrue(keys.any { it.id == key1.id })
+        assertTrue(keys.any { it.id == key2.id })
+    }
+
+    @Test
+    fun delete(): Unit = runBlocking {
+        val key1 = provider.create("foo")
+        assertNotNull(provider.fetch(key1.id))
+        provider.delete(key1.id)
+        assertFailsWith<ServiceException> {
+            provider.fetch(key1.id)
+        }
+    }
+
+    @Test
+    fun delete_nonExisitingKeyNoOps(): Unit = runBlocking {
+        // worth creating a more specific exception?
+        provider.create("foo")
+        val keys = provider.fetchAll()
+        assertEquals(1, keys.size)
+        provider.delete(ObjectId.create())
+        val keysAfterInvalidDelete = provider.fetchAll()
+        assertEquals(1, keysAfterInvalidDelete.size)
+    }
+
+    @Test
+    fun enable(): Unit = runBlocking {
+        val key = provider.create("foo")
+        provider.disable(key.id)
+        assertFalse(provider.fetch(key.id).enabled)
+        provider.enable(key.id)
+        assertTrue(provider.fetch(key.id).enabled)
+    }
+
+    @Test
+    fun enable_alreadyEnabled() = runBlocking {
+        val key = provider.create("foo")
+        provider.disable(key.id)
+        assertFalse(provider.fetch(key.id).enabled)
+        provider.enable(key.id)
+        assertTrue(provider.fetch(key.id).enabled)
+        provider.enable(key.id)
+        assertTrue(provider.fetch(key.id).enabled)
+    }
+    @Test
+    fun enable_nonExistingKeyThrows(): Unit = runBlocking {
+        assertFailsWith<ServiceException> {
+            provider.enable(ObjectId.create())
+        }
+    }
+
+    @Test
+    fun disable() = runBlocking {
+        val key = provider.create("foo")
+        provider.disable(key.id)
+        assertFalse(provider.fetch(key.id).enabled)
+    }
+
+    @Test
+    fun disable_alreadyDisabled() = runBlocking {
+        val key = provider.create("foo")
+        provider.disable(key.id)
+        assertFalse(provider.fetch(key.id).enabled)
+        provider.disable(key.id)
+        assertFalse(provider.fetch(key.id).enabled)
+    }
+
+    @Test
+    fun disable_nonExistingKeyThrows(): Unit = runBlocking {
+        assertFailsWith<ServiceException> {
+            provider.disable(ObjectId.create())
+        }
+    }
```

Comment:
Missing tests for what happens when use is logged out. This is e.g. the tests from Java:

```
   @Test
    fun callMethodWithLoggedOutUser() {
        user.logOut()
        for (method in Method.values()) {
            try {
                when(method) {
                    Method.CREATE -> provider.create("name")
                    Method.FETCH_SINGLE -> provider.fetch(ObjectId())
                    Method.FETCH_ALL -> provider.fetchAll()
                    Method.DELETE -> provider.delete(ObjectId())
                    Method.ENABLE -> provider.enable(ObjectId())
                    Method.DISABLE -> provider.disable(ObjectId())
                }
                fail("$method should have thrown an exception")
            } catch (error: AppException) {
                assertEquals(ErrorCode.SERVICE_UNKNOWN, error.errorCode)
            }
        }
    }
```

Reply:
Also missing tests for accessing `user` and `app` properties.